src/FshConversionProvider.ts

```diff
@@ -0,0 +1,263 @@
+        if (fileUri.fsPath.startsWith(configdir)) {
+          sushiConfigUri = file;
+        }
+      });
+      resolve(sushiConfigUri);
+    });
+  });
+}
+
+function getCanonicalFromSushiConfig(config: SushiConfiguration): string {
+  // try to get canonical: if there's more than one, use the first one that is recognized
+  return config.canonical !== undefined ? config.canonical : 'http://example.org';
+}
+
+function getVersionFromSushiConfig(config: SushiConfiguration): string {
+  // try to get fhirVersion: if there's more than one, use the first one that is recognized
+  let fhirVersion = '4.0.1';
+  const listedVersions = Array.isArray(config.fhirVersion)
+    ? config.fhirVersion
+    : [config.fhirVersion];
+  fhirVersion = listedVersions
+    .map(version => {
+      const versionMatch = version?.match(/^#?(\S*)/);
+      if (versionMatch) {
+        return versionMatch[1];
+      } else {
+        return null;
+      }
+    })
+    .find(version => /current|4\.0\.1|4\.[1-9]\d*\.\d+|[56]\.\d+\.\d+/.test(version));
+
+  return fhirVersion;
+}
+
+function getDependenciesFromSushiConfig(
+  config: SushiConfiguration
+): { packageId: string; version: string }[] {
+  //try to get dependencies: more or less doing SUSHI's importConfiguration.parseDependencies
+  let parsedDependencies: { packageId: string; version: string }[] = [];
+  if (config.dependencies) {
+    parsedDependencies = Object.entries(config.dependencies).map(
+      ([packageId, versionOrDetails]) => {
+        if (typeof versionOrDetails === 'string' || typeof versionOrDetails === 'number') {
+          return { packageId, version: `${versionOrDetails}` };
+        } else if (versionOrDetails == null) {
+          return { packageId, version: undefined };
+        } else {
+          return {
+            packageId,
+            version: versionOrDetails.version ? `${versionOrDetails.version}` : undefined
+          };
+        }
+      }
+    );
+  }
+  return parsedDependencies;
+}
+
+export async function findMatchingFSHResourcesForProject(configFileUri: Uri): Promise<string[]> {
+  const fshResources: string[] = [];
+  const configdir = dirname(configFileUri.fsPath) + path.sep;
+  const files: Uri[] = await workspace.findFiles('**/*.{fsh}');
+
+  for (const file of files) {
+    if (file.fsPath.startsWith(configdir)) {
+      //Found FSH resource belonging to project.
+      const contents = await workspace.fs.readFile(file);
+      const decoder = new TextDecoder();
+      const decodedContents = decoder.decode(contents);
+      fshResources.push(decodedContents);
+    }
+  }
+
+  return fshResources;
+}
+
+export async function findMatchingJsonResourcesForProject(configFileUri: Uri): Promise<string[]> {
+  const jsonResources: string[] = [];
+  const configdir = dirname(configFileUri.fsPath) + path.sep;
+  const files: Uri[] = await workspace.findFiles('**/*.{json}');
+
+  for (const file of files) {
+    if (file.fsPath.startsWith(configdir)) {
```

Comment:
I see, but how to make a difference between resources that are useful during conversion and the ones that are not. The non-fir resources are automatically ignored as they fail to convert as you mentioned. As how I use it currently is when there is a duplicate id either I made a mistake in naming a FHIR resource and I will have to rename it, or it is from a generated FHIR JSON like the output folder of IGPublisher. In that case I remove those generated files. I am hesitant to have certain folders automagically skipped. 

What are your thoughts on this?

Reply:
It sounds like it would not be realistic to try to have special checks for these JSON files. I'm fine with the current implementation, where non-FHIR resources are ignored after failing to convert.